yt/visualization/image_writer.py

```diff
@@ -413,14 +417,11 @@ def write_projection(data, filename, colorbar=True, colorbar_label=None,
         suffix = '.png'
         filename = "%s%s" % (filename, suffix)
     mylog.info("Saving plot %s", filename)
-    if suffix == ".png":
-        canvas = FigureCanvasAgg(fig)
-    elif suffix == ".pdf":
+    if suffix == ".pdf":
```

Comment:
[`get_image_suffix(filename)`](https://github.com/yt-project/yt/blob/master/yt/funcs.py#L811) returns `['.png', '.eps', '.ps', '.pdf', '']` thus, it will never execute the original `else` block.

Reply:
Ah, fair enough